sdk/core/azure_core_amqp/src/error.rs

```diff
@@ -32,7 +32,7 @@ pub enum AmqpErrorKind {
     TransportImplementationError(Box<dyn std::error::Error + Send + Sync>),
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
If this is intended to be fully equatable, yo ushould also derive `Eq`. Generally that will be the case. If you only want *truly* `PartialEq`, it's more often `impl`'d because you have to say exactly what is compared. `Eq` is a marker trait - a super trait of `PartialEq` that means "fully equatable", basically.

Reply:
Actually it was a part of an experiment for my tests trying to compare two ErrorKind enumerations.

Reply:
So removing it :)